### PR TITLE
convert the dht commands to the new commands lib

### PR DESCRIPTION
**Pull request by Stebalien**
Also, stop accepting variadic arguments in single argument commands.

Also, -150loc :)

License: MIT
Signed-off-by: Steven Allen <steven@stebalien.com>

### CODE REVIEW

**Comment by Stebalien** [comment]
@magik6k this should fix the bug you told me about on IRC.

File: core/commands/dht.go [hunk 1]
```diff
@@ -145,14 +113,14 @@ var findProvidersDhtCmd = &cmds.Command{
 	},
 
 	Arguments: []cmdkit.Argument{
-		cmdkit.StringArg("key", true, true, "The key to find providers for."),
+		cmdkit.StringArg("key", true, false, "The key to find providers for."),
```
Comment (Kubuxu) [inline core/commands/dht.go]: Why disable variadic here?

Comment (Kubuxu) [reply core/commands/dht.go]: On the other hand, it makes sense as it never worked. Ok.

**Comment by Stebalien** [comment]
Unfortunately, we should iron out https://github.com/ipfs/go-ipfs-cmds/issues/65 first.

**Comment by eingenito** [comment]
@Stebalien I'm closing this because I think it's been completed in https://github.com/ipfs/go-ipfs/pull/5671. Please reopen if that's not correct.

**Pull request closed without merging**